tensorflow_datasets/core/dataset_builder.py

```diff
@@ -284,7 +284,17 @@ def _checksums_path(cls) -> ReadOnlyPath:
     # Used:
     # * To load the checksums (in url_infos)
     # * To save the checksums (in DownloadManager)
-    return cls.code_path.parent / "checksums.tsv"
+    #Checking whether the dataset follow the folder structure
+    path = str(cls.code_path.parent)
+    if path.split('\\')[-1] == cls.name:
```

Comment:
Try to work directly using Pathlib API

Like `cls.code_path.parent.name == cls.name`